### DISCUSSION TITLE
Add SVG asset to public directory for Vite app

### DISCUSSION THREAD

**Discussion by Pettor** [vercel/turborepo#5046] [Help]
My Tubrorepo structure has a web-app that uses Vite to run and a shared UI from the package folder like this:

```
- apps/web
- package/shared-ui
```

The shared UI has a StyledComponent that imports an SVG and use it as a background (just an example, so not the full component):

```
import SomeSvg from "./svg/SomeSvg.svg";

export const SomeComponent = styled("div")(() => ({ backgroundImage: `url`${SomeSvg}` }));
```

The SVG file will end up in the dist of `shared-ui` but my Vite app will never include `SomeSvg.svg` in the public folder and therefore allow the component to fetch the SVGs as a background image. If I copy the svg files to the public folder in my web-app everything will work.

Is there a way to build packages and basically say "these are public assets and should be included in the build as such"?

Thanks 🙏

**Comment by anthonyshew** [comment]
We don't have a Turborepo-native solution for this today and I don't know of anything in the JS/TS ecosystem that solves for this out-of-the-box. You can certainly write a Node.js script to spread your static assets around into your application's public directories prior to builds.

That said, I'd be interested in seeing if we could figure out a Turbo native solution for this.

**Reply by zigang93** [reply]
> We don't have a Turborepo-native solution for this today and I don't know of anything in the JS/TS ecosystem that solves for this out-of-the-box. You can certainly write a Node.js script to spread your static assets around into your application's public directories prior to builds.
> 
> That said, I'd be interested in seeing if we could figure out a Turbo native solution for this.

Does this have any roadmap or idea yet?

**Reply by anthonyshew** [reply]
No, we haven't spent time thinking about this, and it's not clear at this point that this is something that _should_ be solved with Turborepo.


In general, folks who have a need for this should really be using a blob storage solution, persisting the assets outside of their Git repo. Static assets are large blobs, which degrade Git performance quickly.

Still, if you do insist on storing the assets in your Git repo, a bundler that handles static assets could be used to bundle to something Node.js can export from a package. Again, this goes against first principles, but you can still do it.

**Reply by zigang93** [reply]
I see.. We are using private network when deploy the apps so the only option to host at aws s3 is not allowed. I am still finding another workaround

**Comment by AabhasDhaubanja** [comment]
I don't know if this is the right way to do it but what I did was create a separate `assets` package. I named the package **@repo/assets**. Then in the component use it like:

```
import SomeSvg from "@repo/assets/SomeSvg.svg";

export const SomeComponent = styled("div")(() => ({ backgroundImage: `url`${SomeSvg}` }));
```

**Reply by maciejbaczynski** [reply]
@AabhasDhaubanja

Can you show how your @repo/assets looks like?

**Reply by userAugustos** [reply]
The only problem on this is the difference between what is asset and what is on public, and how the bundler will treat this

https://stackoverflow.com/questions/44643041/do-i-store-image-assets-in-public-or-src-in-reactjs

And if you wanna put something on your public folder, let's say from your app (which calls your package) you could do:


```ts
declare module '@public/*' {
  const value: any;
  export default value;
}
```

and them:

```tsx
import Image from '@public/image.png';
```

This will at least allow your inteliSense to work (LSP) and on build and dev no problems as well.

For build you gotta add this to your alias:

```ts

resolve: {
      alias: [
          {
             find: '@public',
             replacement: './public'
          },
      ]
},
```

Idk if it's the best solution, but it works well, what is public is used as it should, and the IDE adapts right to it.

**Comment by JoyShaheb** [comment]
i solved the issue like this for my use case
```
import { ICustomerDetails } from "../_Types";
import someImageHere from "../images/Company/ABCD/an-image.jpg";

export const CUSTOMER_DETAILS: ICustomerDetails[] = [
  {
    name: "ABCD",
    backgroundImage: someImageHere,
  },
  {
    name: "DEMO",
  },
  {
    name: "BDAJDGAwey",
  },
];

```

There might be other better solutions out there which i might not know, please feel free to reply to this and educate me 🙏
